packages/safe-promise-all/index.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const safePromiseAll = (arrP) =>
+    Promise.allSettled(arrP).then((results) => {
+        const errors = results
+            .filter(({ status }) => status === 'rejected')
+            .map(({ reason }) => reason)
+            .flat();
+
+        if (errors.length > 0) {
+            return [errors];
```

Comment:
@allanchau, it should always return errors and values - that's the point of using `allSettled`.